### PR TITLE
323 add error when mail already exists

### LINKED ISSUE #323: Add error if someone tries to sign up with an email we already have in the DB (both for mentor and mentee)

### PR DESCRIPTION
**Pull request by wrc9817**

### CODE REVIEW

**Comment by wrc9817** [comment]
The form of the alert message can be in any other form. We may decide it later.

**Comment by jpabadir** [comment]
This PR has merge conflicts. Can you try to fix them please?

File: src/front-end/src/components/MenteeForm.js
Comment (jpabadir) [inline src/front-end/src/components/MenteeForm.js]: I'd remove this, so as not to clutter the inspector with logs.

File: src/front-end/src/components/MentorForm.js
Comment (jpabadir) [inline src/front-end/src/components/MentorForm.js]: I'd remove this, so as not to clutter the inspector with logs.

File: src/front-end/src/components/MentorForm.js
Comment (jpabadir) [inline src/front-end/src/components/MentorForm.js]: Firebase already handles this for us in authentication. I made a commit to transfer our code to use their built-in system (it's less verbose and more efficient). Feel free to look at the code and ask if you have any questions. 

On the mentee side, however, your code is valid since we're not using authentication.

**Review by jpabadir — CHANGES_REQUESTED** [review]
Good PR, you can read my comments for feedback. Two things needed before merging: 

- Resolve the merge conflicts
- Remove the log statement, if you agree with me. If you don't you can keep it there

File: src/front-end/src/components/MenteeForm.js
Comment (wrc9817) [reply src/front-end/src/components/MenteeForm.js]: Sure!

File: src/front-end/src/components/MentorForm.js
Comment (wrc9817) [reply src/front-end/src/components/MentorForm.js]: KK I will leave it how it was like.

File: src/front-end/src/components/MentorForm.js
Comment (wrc9817) [reply src/front-end/src/components/MentorForm.js]: Sure!

File: src/front-end/src/components/MentorForm.js
Comment (jpabadir) [inline src/front-end/src/components/MentorForm.js]: You probably deleted this when fixing the merge conflicts. When you fix the conflicts, you need to make sure you keep both your functionality and the functionality of the branch with which you have a conflict. Here, if we merge this, we are deleting the confirm password functionality, which is present in develop, and should therefore be kept. Please try again to resolve the merge conflict in a way which adds your desired functionality without removing current functionality we have. 

You can watch [this video](https://www.youtube.com/watch?v=JtIX3HJKwfo) about resolving merge conflicts.

**Review by jpabadir — CHANGES_REQUESTED** [review]

**Review by aleksssk — APPROVED** [review]
Looks good after changes are fixed.

File: src/front-end/src/components/MentorForm.js
Comment (katmartel) [reply src/front-end/src/components/MentorForm.js]: @wrc9817 If you're not able to resolve JP's comment, let me know and I can fix the form tomorrow morning (well, morning for me, evening for you) :)

Comment (wrc9817) [reply src/front-end/src/components/MentorForm.js]: @jpabadir Thanks for the feedback. I did not realize that.
@KatrinaMart I think I'm good to deal with it. Thank you tho!

Comment (katmartel) [reply src/front-end/src/components/MentorForm.js]: @wrc9817 I can see that the confirm password field has been added again, but the changes that I had pushed to develop as part of PR #301 have been removed (likely to have been accidentally deleted while trying to fix the merge conflict(s)). If you run the program, you can see that compared to the MenteeForm (which has the changes from PR #301), the MentorForm looks drastically different (for ex. the imported TimeZone library should no longer be on the MentorForm). 
Please let me know if this makes sense, and if you'll be able to incorporate the work that you've done without overriding what was merged in develop from PR #301

**Comment by katmartel** [comment]
@wrc9817 The PR now includes the existing functionality from develop with the additional features that you've added in this branch.
@jpabadir When you get the chance, please review this PR to see if the changes that you've requested have been incorporated. 
If approved, once we figure out the failing jest test, this PR should be good to merge.

**Comment by wrc9817** [comment]
> @wrc9817 The PR now includes the existing functionality from develop with the additional features that you've added in this branch.
> @jpabadir When you get the chance, please review this PR to see if the changes that you've requested have been incorporated.
> If approved, once we figure out the failing jest test, this PR should be good to merge.

**Comment by wrc9817** [comment]
> > @wrc9817 The PR now includes the existing functionality from develop with the additional features that you've added in this branch.
> > @jpabadir When you get the chance, please review this PR to see if the changes that you've requested have been incorporated.
> > If approved, once we figure out the failing jest test, this PR should be good to merge.

Thank you for doing this! Really Appreciate it.

**Review by jpabadir — APPROVED** [review]

**Pull request merged**
